### ISSUE TITLE
Reloading avatars often when they hit max_upload_size and creating a lot of traffic

### ISSUE THREAD

**Issue by benvei** [matrix-org/synapse#9453]
### Description

I'm running matrix-synapse on Debian.
I've set **max_upload_size: 10M** in my homeserver config.

I've noticed that my instance creates a LOT of incomming traffic and traced it down room / user avatars being loaded.
Synapse puts out a lot of logs about to big files:

`2021-02-19 18:59:31,879 - synapse.http.matrixfederationclient - 997 - WARNING - GET-7558 - {GET-O-5042} [xxx.xxx] Requested file is too large > 10485760 bytes
2021-02-19 18:59:31,884 - synapse.http.server - 86 - INFO - GET-7558 - <XForwardedForRequest at 0xxxxxxxxxx method='GET' uri='/_matrix/media/r0/thumbnail/xxx.xxx/xxxxxxxxxxx?width=37&height=37&method=crop' clientproto='HTTP/1.0' site='8008'> SynapseError: 502 - Requested file is too large > 10485760 bytes
`

The issue is that matrix-synapse tries to load the avatar every time the room is accessed our you write with a user that has an avatar >10mb. This actuall creates a lot of traffic since synapse tries to download it and aborts it as soon as it hits 10mb and tries it again. It does not save the state, that the file is larger than 10mb. In extreme scenarios it tries to download it multiple times per minute (e.g. if the user sends multiple messages).

I've set the upload_size limit to protect my bandwidth since I'm hosting on quite a low of bandwidth. But in that case it did the opposite.

### Steps to reproduce

- install synapse
- set max_upload_size: 10M
- join a room with an avatar >10mb
- click on the room a few times (avatar gets loaded every time)
- check logfiles

<!--
Describe how what happens differs from what you expected.

If you can identify any relevant log snippets from _homeserver.log_, please include
those (please be careful to remove any personal or private data). Please surround them with
``` (three backticks, on a line on their own), so that they are formatted legibly.
-->

### Version information

<!-- IMPORTANT: please answer the following questions, to help us narrow down the problem -->

<!-- Was this issue identified on matrix.org or another homeserver? -->
- **Homeserver**


<!--
 What version of Synapse is running?

You can find the Synapse version with this command:

$ curl http://localhost:8008/_synapse/admin/v1/server_version

(You may need to replace `localhost:8008` if Synapse is not configured to
listen on that port.)
-->
- **Version**: 1.27.0+buster1

- **Install method**: package manager
<!-- examples: package manager/git clone/pip  -->

- **Platform**: Debian
<!--
Tell us about the environment in which your homeserver is operating
distro, hardware, if it's running in a vm/container, etc.
-->

Matrix-Synapse running on Debian on a LXC container.

**Comment by clokep** [comment]
I agree that it seems a bit silly that we keep attempting to fetch something that will fail. I think this will be a tiny bit better in v1.28.0 (as #9421 will cause Synapse to more closely adhere to the maximum limit instead of downloading and then discarding data).

If there's only a couple of rooms causing an issue a workaround might be to up the limit temporarily, load those rooms, then reduce the limit again.
